File: backend/app/schemas/log.py
"""Access Log Pydantic schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class AccessLogRead(BaseModel):
    """Schema for reading access logs."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int | None
    action: str
    resource: str | None
    ip_address: str | None
    status: str
    details: str | None
    timestamp: datetime


class AccessLogList(BaseModel):
    """Paginated list of access logs."""

    logs: list[AccessLogRead]
    total: int

File: frontend/src/types/log.ts
export interface AccessLog {
  id: number
  user_id: number | null
  action: string
  resource: string | null
  ip_address: string | null
  user_agent: string | null
  status: 'success' | 'failure' | string
  details: string | null
  timestamp: string
}

export interface AccessLogList {
  logs: AccessLog[]
  total: number
}
